docs/user-defined-functions-c#.md

```diff
@@ -0,0 +1,43 @@
+# User-Defined Functions - C#
+This documentation contains user-defined function (UDF) examples. It shows how to define UDFs and how to use UDFs with Row objects as examples.
+
+## Pre-requisites:
+Install Microsoft.Spark.Worker. When you want to execute a C# UDF, Spark needs to understand how to launch the .NET CLR to execute this UDF. Microsoft.Spark.Worker provides a collection of classes to Spark that enable this functionality. Please see more details at [how to install Microsoft.Spark.Worker](https://docs.microsoft.com/en-us/dotnet/spark/tutorials/get-started#5-install-net-for-apache-spark) and [how to deploy worker and UDF binaries](https://docs.microsoft.com/en-us/dotnet/spark/how-to-guides/deploy-worker-udf-binaries).
+
+## UDF that takes in Row objects
+
```

Comment:
Can we add some sentences providing context to this example? 

For instance, as a reader, I have the following questions:
- When would I use a UDF that takes in Row objects (as opposed to other types of UDFs)? 
- Do all UDFs just take in or return Row objects (since that's all that is shown in this doc)?
- What is the goal of this code? What calculation or filtering is it performing and why?
- What would be the output of this code?
- Is this the only way to define UDFs (using `Func<> myUdf = Udf<>(...)`)? What about `spark.Udf().Register<>...`?

Reply:
Thanks for your suggestion! I was looking at UDF docs [here](https://docs.databricks.com/spark/latest/spark-sql/udf-python.html). I am not sure how much detail we want to go with this intro guide. Should we just consider this as a `using UDF with Row objects` readme file or UDF tutorial? This goes with your previous question also.